src/alr/alr-commands-withing.adb

```diff
@@ -357,6 +407,20 @@ package body Alr.Commands.Withing is
                 & " specified with --branch. Use `alr update` to refresh the"
                 & " tracking pin contents.")
        .New_Line
+       .Append ("* Adding dependencies with shared storage:")
```

Comment:
Is `alr with --shared libhello` also adding `libhello` as a dependency or only setting the expected sharing?

I think a dedicated command for this feature might make more sense.

Reply:
It's currently doing both. I'm also not really happy with it, but couldn't find at the time an entirely satisfying interface.

One possibility is reusing `alr config`, which already has the `--global` needed to differentiate global/workspace settings. So it could be:

```
alr config --share[=yes|no|reset] <crate> [--global]
```

I'm open to suggestions.

Reply:
`alr share [--global|--local] [--yes|--no] <crate>` ?

Default to `--local` and `--yes`.

And a `--info` to show the local and global shares.